client/src/components/common/Debounce.tsx

```diff
@@ -0,0 +1,40 @@
+import {useCallback, useEffect, useState} from "react";
+import { UseDebounceProps } from "../../types/Debouncer/types";
+export default function useDebounce<T>({ object, delay = 1000}: UseDebounceProps<T>){
+  const [value, setValue] = useState<T | undefined>()
+  const [isBouncing, setBouncing] = useState(true)
+
+  const computeDelay = useCallback(() => {
+    if(!object){
```

Comment:
u might need to make the rules less strict to support other more complex types

Reply:
Ok